ui/mirage/factories/variable.js

```diff
@@ -22,4 +22,19 @@ export default Factory.extend({
       }
     );
   },
+
+  afterCreate(variable, server) {
+    if (!variable.namespace) {
+      const namespace = server.db.namespaces.length
+        ? pickOne(server.db.namespaces).id
+        : null;
+      variable.update({
+        namespace,
+      });
+    } else {
+      variable.update({
+        namespace: variable.namespace,
```

Comment:
Noticing a bug here: the provided namespace and the returned namespace don't match in mirage
![image](https://user-images.githubusercontent.com/713991/177795377-07677aab-a2a9-4d4d-b252-1baf2ff913fa.png)

Reply:
That's a good catch, we probably have a similar bug in the `jobs` factory because that's where I copied this code from.